pandas/tests/indexes/test_numpy_compat.py

```diff
@@ -84,8 +82,18 @@ def test_numpy_ufuncs_basic(index, func):
 
         tm.assert_index_equal(result, exp)
         if type(index) is not Index or index.dtype == bool:
-            # i.e NumericIndex
-            assert isinstance(result, Float64Index)
+            assert type(result) is NumericIndex
+            if is_complex_dtype(index):
+                assert result.dtype == "complex64"
+            elif index.dtype in ["bool", "int8", "uint8"]:
+                # TODO: find out when index.dtype is float16 vs float32 (platform issue)
```

Comment:
is this still relevant or was float16 disallowed?

Reply:
Yeah, that's not relevant anymore. I've updated.